ui/src/plugin/PluginDetailView.tsx

```diff
@@ -0,0 +1,259 @@
+            const response = await axios.get(
+                `${config.get('url')}plugin/${this.state.pluginInfo!.id}/config`
+            );
+            this.setState({currentConfig: response.data});
+        }
+    }
+
+    private async refreshDisplayer() {
+        if (this.state.pluginInfo!.capabilities.indexOf('displayer') !== -1) {
+            const response = await axios.get(
+                `${config.get('url')}plugin/${this.state.pluginInfo!.id}/display`
+            );
+            this.setState({displayText: response.data});
+        }
+    }
+
+    public render() {
+        return this.state.pluginInfo ? (
+            <DefaultPage title={this.state.pluginInfo.name} hideButton={true} maxWidth={1000}>
+                <PanelWrapper name={'Plugin Info'} icon={Info}>
+                    <PluginInfo pluginInfo={this.state.pluginInfo} />
+                </PanelWrapper>
+                {this.state.pluginInfo.capabilities.indexOf('configurer') !== -1 ? (
+                    <PanelWrapper
+                        name={'Configurer'}
+                        description={'This is the configuration panel for this plugin.'}
+                        icon={Build}
+                        refresh={this.refreshConfigurer.bind(this)}>
+                        <ConfigurerPanel
+                            pluginInfo={this.state.pluginInfo}
+                            initialConfig={
+                                this.state.currentConfig ? this.state.currentConfig : 'Loading...'
+                            }
+                            save={async (newConfig) => {
+                                await axios.post(
+                                    `${config.get('url')}plugin/${
+                                        this.state.pluginInfo!.id
+                                    }/config`,
+                                    newConfig,
+                                    {headers: {'content-type': 'application/x-yaml'}}
+                                );
+                                await this.refreshFeatures();
+                            }}
+                        />
+                    </PanelWrapper>
+                ) : null}{' '}
+                {this.state.pluginInfo.capabilities.indexOf('displayer') !== -1 ? (
+                    <PanelWrapper
+                        name={'Displayer'}
+                        description={'This is the information generated by the plugin.'}
+                        refresh={this.refreshDisplayer.bind(this)}
+                        icon={Subject}>
+                        <DisplayerPanel
+                            pluginInfo={this.state.pluginInfo}
+                            displayText={this.state.displayText || 'Loading...'}
+                        />
+                    </PanelWrapper>
+                ) : null}
+            </DefaultPage>
+        ) : null;
+    }
+}
+
+class PanelWrapper extends Component<{
```

Comment:
This can also be converted to a stateless functional compoenent.

If the props have more than one entry, then I'd prefer an separate interface for the props. F.ex.:
```typescript
interface PanelWrapperProps {
    name: string;
    description?: string;
    refresh?: () => Promise<void>;
    icon?: React.ComponentType;
}
const PanelWrapper: React.SFC<PanelWrapperProps> = ({name, description, refresh, icon}) => ...
```